### PR TITLE
feat(build): added support for publishing api docs and showcase using Travis

### PR DESCRIPTION
**Pull request by dsebastien**
Issues closed: #282

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/NationalBankBelgium/stark/blob/master/CONTRIBUTING.md#-commit-message-guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[x] Build related changes
[x] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?
We generate our docs but don't publish them

Issue Number: #282 

## What is the new behavior?
We publish the api docs and the showcase on GitHub pages.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information
The solution is not complete yet, because other issues need to be tackled for this to be effective; for example #415

### CODE REVIEW

**Comment by dsebastien** [comment]
To actually test this, we'll need to create a release.
I also noticed that the current npm packages we publish still refer to ...alpha.2 as the "next", which doesn't make much sense if the next will actually be at least alpha.3 ^^

File: gh-deploy.sh [hunk 1]
```diff
@@ -0,0 +1,326 @@
...
+SOURCE_BRANCH="master"
+TARGET_BRANCH="gh-pages"
+COMMIT_HASH=`git rev-parse --verify HEAD`
+
+TARGET_REPO="git@github.com:NationalBankBelgium/stark.git"
+EXPECTED_REPO_SLUG="NationalBankBelgium/stark"
+EXPECTED_NODE_VERSION="8"
+
+COMMIT_AUTHOR_USERNAME="TravisCI"
+COMMIT_AUTHOR_EMAIL="seb@dsebastien.net"
```
Comment (SuperITMan) [inline gh-deploy.sh]: Should not we create a specific email address (NBB) for Stark on GitHub ?

**Comment by SuperITMan** [comment]
Actually, when we do a release, I think we should update the version to the next patch/minor one automatically 😊

File: gh-deploy.sh (hunk 1 above)
Comment (dsebastien) [reply gh-deploy.sh]: Yes but no :)

**Comment by christophercr** [comment]
@dsebastien I tested again this and indeed the SSH key wasn't used when doing the git clone/push.

So I used the other alternative you mentioned, and that works, this is the log of the successful build of my fork: https://travis-ci.org/christophercr/stark/jobs/393357177

I've adapted my fork so you can see the changes I did compared to what you have (but you need to rebase yours with the latest changes from the master): https://github.com/dsebastien/stark/compare/feature/ghdeploy...christophercr:feature/ghdeploy

**Comment by dsebastien** [comment]
Great!

Don’t hesitate to just adapt your fork and PR/merge that in upstream :)

**Comment by christophercr** [comment]
Closed in favor of this PR #445 which contains the final working solution.

**Pull request closed without merging**